CI: dependabot bumps of `@typescript/native-preview` fail frozen-lockfile because `website/package.json` pins it outside the catalog

## Summary

Every dependabot PR that bumps `@typescript/native-preview` (currently #1911) fails all CI jobs within ~20 seconds, at the very first `pnpm install --frozen-lockfile` step:

```
ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up to date with <ROOT>/website/package.json
```

No build or test ever runs — the failure is purely a lockfile consistency problem, not a typescript-go behavioral break (that part remains untested until install succeeds).

## Root cause

`@typescript/native-preview` is declared in two places on `master`:

| Location | Declaration |
|---|---|
| `pnpm-workspace.yaml` → `catalogs.typescript` | `7.0.0-dev.20260527.2` (consumed via `catalog:typescript` by `packages/*`, `tests/test-interface`, `benchmark`) |
| `website/package.json` (workspace member) | hardcoded `"7.0.0-dev.20260429.1"` |

Dependabot's pnpm-catalog support handles the catalog path fine: it rewrites the catalog in `pnpm-workspace.yaml` plus the matching catalog/snapshot entries in `pnpm-lock.yaml`. It also rewrites the hardcoded version in `website/package.json`. What it does **not** do is regenerate the `website` importer block of `pnpm-lock.yaml` — in #1911 that block still records `specifier: 7.0.0-dev.20260429.1` while `website/package.json` now says `7.0.0-dev.20260609.1`, which is exactly the mismatch `--frozen-lockfile` rejects.

So any future tsgo bump will keep failing the same way as long as the out-of-catalog pin exists.

## Suggested fix

Switch the website to the catalog and refresh the lockfile once:

```diff
// website/package.json
-    "@typescript/native-preview": "7.0.0-dev.20260429.1",
+    "@typescript/native-preview": "catalog:typescript",
```

then `pnpm install` and commit `pnpm-lock.yaml`. With a single source of truth in the catalog, dependabot bumps only touch `pnpm-workspace.yaml` + lockfile catalog entries, which it already keeps consistent — the equivalent bump PR on ttsc (samchon/ttsc#212, catalog-only declaration) installs cleanly.

`experiments/smoke/package.json` also hardcodes `7.0.0-dev.20260510.1`, but it is not a workspace member (`pnpm-workspace.yaml` `packages:` doesn't include `experiments/*`), so it doesn't affect the lockfile; aligning it is optional housekeeping.

## Unrelated but adjacent, for the record

samchon/ttsc#212 (same tsgo bump on the ttsc side) showed a `typia.yml` failure in `tests/test-typia-exact-optional`, which looked like the bump breaking typia. It wasn't: that workflow clones typia `master` at run time, and the failing run (2026-06-09 12:44 UTC) happened to clone the window between 76c981ad4 (`fix(typia): respect exact optional undefined`, which introduced those tests) and b15e2991a (`fix(typia): restore exact optional undefined handling`, which fixed them). The npm tsgo version isn't even consumed by that test path — ttsc's compiler is pinned via `go.mod`, unchanged in that PR.

--- a/.codex/skills/development/SKILL.md
+++ b/.codex/skills/development/SKILL.md
@@ -70,7 +70,7 @@ Match the scope of the command to the scope of the change. Report any command yo
 - Touched one test workspace → `pnpm --filter ./tests/<name> start`.
 - Touched the transform, descriptors, or a shared package → `pnpm test`.
 - Touched packaging → `pnpm package:tgz` from `experiments/tarballs` (stages tarballs the website consumes) and try a clean install. Don't commit or hand-edit the staged tarballs.
-- Touched `website/src/content/docs/**` or anything else under `website/` → `cd website && npm install && npm run build` to validate MDX, nav, and tarball pickup.
+- Touched `website/src/content/docs/**` or anything else under `website/` → root `pnpm install`, then `pnpm run build` inside `website/` to validate MDX, nav, and tarball pickup. The website is a pnpm workspace member consuming `catalog:` versions, so a standalone `npm install` inside `website/` cannot resolve its dependencies.
 
 ## Change Integrity
 

--- a/experiments/smoke/package.json
+++ b/experiments/smoke/package.json
@@ -12,7 +12,7 @@
     "typia": "file:../tarballs/typia.tgz"
   },
   "devDependencies": {
-    "@typescript/native-preview": "7.0.0-dev.20260510.1",
+    "@typescript/native-preview": "7.0.0-dev.20260527.2",
     "ttsc": "^0.15.1"
   }
 }

--- a/website/package.json
+++ b/website/package.json
@@ -63,7 +63,7 @@
     "@types/node": "^18.11.10",
     "@types/prettier": "^3.0.0",
     "@types/react": "^18.0.35",
-    "@typescript/native-preview": "7.0.0-dev.20260429.1",
+    "@typescript/native-preview": "catalog:typescript",
     "gh-pages": "^5.0.0",
     "next-sitemap": "^4.2.3",
     "pagefind": "^1.3.0",

